### PR TITLE
[revest] Created Revest Staking System strategy 

### PR DESCRIPTION
**Pull request by RobAnon**
A custom strategy proposed to offer support for snapshots taken through the Revest staking system. Calculates who has what representational power based on the size of their staked position into Revest and the length it is locked up for, all off of what FNFTs they possess and the voting power each FNFT represents on its own

### CODE REVIEW

File: src/strategies/revest/index.ts [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    );
+    rewards = rewards[0][0];
+    let allocLP = await multicall(
+      network,
+      _provider,
+      abi3,
+      [""].map((id: any) => [rewards, 'totalLPAllocPoint', []]),
+      {blockTag}
+    );
+    let allocToken = await multicall(
```
Comment (ChaituVR) [inline src/strategies/revest/index.ts:126]: Hi @RobAnon There is a limit on the number of requests being sent (multicall/subgraph/fetch), It is limited to 5 calls 

https://github.com/snapshot-labs/snapshot-strategies#checklist-for-adding-a-new-strategy

Can you try to find if there is some alternate way to get this in fewer calls? maybe try sending multiple calls into multicall

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: src/strategies/revest/index.ts (hunk 1 above)
Comment (RobAnon) [reply src/strategies/revest/index.ts:126]: Can I call multiple functions via multicall?

Comment (RobAnon) [reply src/strategies/revest/index.ts]: @ChaituVR I've reduced the number to 5 calls through the use of multicall

File: src/strategies/revest/index.ts [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    });
+    let returnVals = {};
+    Object.keys(finalResult).forEach(element => {
+      returnVals[element] = parseInt(finalResult[element].toString(), 10); 
+    });
+    console.log(returnVals);
+    return returnVals;
+  } catch (err) {
+    return {};
+  }
```
Comment (ChaituVR) [inline src/strategies/revest/index.ts:152]: Hi, It should still throw an error, `{}` may mean there is 0 voting power for all addresses, maybe remove the try-catch, should be good to go :)

**Review by ChaituVR — CHANGES_REQUESTED** [review]

**Comment by RobAnon** [comment]
@ChaituVR Ready to merge

File: src/strategies/revest/index.ts [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+          finalResult[getAddress(account.id)] = score;
+        }
+      }
+    });
+  });
+  let returnVals = {};
+  Object.keys(finalResult).forEach(element => {
+    returnVals[element] = parseInt(finalResult[element].toString(), 10); 
+  });
+  console.log(returnVals);
```
Comment (ChaituVR) [inline src/strategies/revest/index.ts:147]:
```suggestion
```

**Review by ChaituVR — APPROVED** [review]

**Pull request merged**